jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -597,18 +653,46 @@ static Error from(Throwable failure)
          */
         static Chunk next(Chunk chunk)
         {
-            if (chunk == null || chunk instanceof Error)
-                return chunk;
+            if (chunk == null)
+                return null;
+            if (Content.Chunk.isError(chunk))
+                return chunk.isLast() ? chunk : null;
             if (chunk.isLast())
                 return EOF;
             return null;
         }
 
+        /**
+         * @param chunk The chunk to test for an {@link Chunk#getError() error}.
+         * @return True if the chunk is non-null and {@link Chunk#getError() chunk.getError()} returns non-null.
+         */
+        static boolean isError(Chunk chunk)
```

Comment:
Rename to `isFailure()`.

Reply:
@sbordet The names were discussed at length between @lorban and myself.   We did try the renames that you suggest, but they don't really work.

Currently we have `isError` and `getError`, where an "error" can be a failure from `fail(Throwable)` or a warning from `warn(Throwable)`

An alternative is to have `isFailure` and `getFailure`, where a "failure" can be a persistent from `fail(Throwable)` or a either from `faul(Throwable, boolean last)`.  @lorban thoughts?

Eitherway, the language used is more than just these 2 methods and updates need to be coordinated through the javadoc and other methods.

Reply:
@sbordet note that @lorban and I did consider the `fail(Throwable failure, boolean last)` approach, but there are big differences between the semantics of last being true/false.
With last==true, any queued chunks are failed and the source is drained and failed immediately.  With last==false, the "warning" is handled in order with other chunks and no draining occurs.

I guess that can be OK behind the one method with careful javadoc?

Reply:
I vote -0 on `fail(Throwable failure, boolean last)` for the reasons @gregw explained: the last flag completely changes the semantic so I'd prefer to have two separate methods, but could live with a single well-documented one.

Reply:
@lorban thoughts on naming? again?

Reply:
I prefer to stick to what we have: `isError`, `getError` and `fail(Throwable)`. I just wouldn't add `warn(Throwable)` on `Content.Source` yet and only have it on `AsyncContent` for now.

Reply:
@lorban now you say it... I'm liking less the usage of error, fail and warn,  when we could just do fail and javadoc. so I'm ±0 for the current naming.   @sbordet if you do look at this, what are your thoughts now we've explained the dilema?

Reply:
I would stick with fail(Throwable), `Throwable getFailure()` and `isFailed(Chunk)`.